### PR TITLE
Document replace tables for Delta Lake and update section for Iceberg

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

This PR documents the `CREATE OR REPLACE TABLE` and `CREATE OR REPLACE TABLE AS` statements for the Iceberg and Delta lake connectors.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 1]
```diff
@@ -724,6 +725,37 @@ directly or used in conditional statements.
 - `$file_size`
   : Size of the file for this row.
 
+### Replace tables
+
+The connector supports replacing a table, in case the table already exists, as
+an atomic operation. Atomic table replacement creates a new snapshot with the
+new table definition and keeps the existing table history. See
+{ref}`delta-lake-history-table` in metadata tables.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:733]: use markdown syntax for ref link `[](delta-lake-history-table)` .. and optionally with title

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 2]
```diff
@@ -724,6 +725,37 @@ directly or used in conditional statements.
...
+
+To replace a table, use `CREATE OR REPLACE TABLE` and/or `CREATE OR REPLACE
+TABLE AS`. For more information on creating tables, see {doc}`CREATE TABLE
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:736]: use markdown syntax for link
```
[](/sql/create-table)
```
same for others

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 3]
```diff
@@ -724,6 +725,37 @@ directly or used in conditional statements.
...
+
+For example, a table `example_table` can be replaced by a completely new
+definition:
+
+```
+CREATE TABLE example_table (
+    a BIGINT,
+    b DATE,
+    c BIGINT)
+WITH (partitioning = ARRAY['a']);
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:752]: confirm this is valid table property in delta lake

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 4]
```diff
@@ -724,6 +725,37 @@ directly or used in conditional statements.
...
+
+CREATE OR REPLACE TABLE example_table
+WITH (sorted_by = ARRAY['a'])
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:755]: confirm table property

**Review by mosabua — CHANGES_REQUESTED** [review]
Please fix up the links

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 3 above)
Comment (ebyhr) [reply docs/src/main/sphinx/connector/delta-lake.md:752]: It should be `partitioned_by`. 
https://github.com/trinodb/trino/blob/23e5ec5b3d30182548d90636a8fbaec9c82ffcfe/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java#L41

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 4 above)
Comment (ebyhr) [reply docs/src/main/sphinx/connector/delta-lake.md:755]: The connector doesn't support such property.

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 5]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
 - `$file_size`
   : Size of the file for this row.
 
+### Replace tables
+
+The connector supports replacing a table, in case the table already exists, as
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:786]:
```suggestion
The connector supports replacing an existing table as
```

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 6]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+an atomic operation. Atomic table replacement creates a new snapshot with the
+new table definition and keeps the existing table history. See
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:788]:
```suggestion
new table definition as part of the [table history](#delta-lake-history-table).
```

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 7]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+[](#delta-lake-history-table) in metadata tables.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:789]: remove

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 8]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+
+To replace a table, use `CREATE OR REPLACE TABLE` and/or `CREATE OR REPLACE
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:791]:
```suggestion
To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or [`CREATE OR REPLACE AS`](/sql/create-table-as).
```

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 9]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+TABLE AS`. For more information on creating tables, see [](/sql/create-table)
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:792]: Remove from "For more information" onwards

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 10]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+and [](/sql/create-table-as). These statements do not support changing table
+formats defined by the `type` table property.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:794]: remove this . there is no type table property in Trino.

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 11]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+
+After replacement, the table definition is completely new and separate from the
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:796]: Remove .. this is wrong

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 12]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+old table.
+
+For example, a table `example_table` can be replaced by a completely new
+definition:
+
+```
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:802]:
```suggestion
```sql
```

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 13]
```diff
@@ -780,6 +781,36 @@ directly or used in conditional statements.
...
+CREATE TABLE example_table (
+    a BIGINT,
+    b DATE,
+    c BIGINT)
+WITH (partitioning = "partitioned_by");
+
+CREATE OR REPLACE TABLE example_table
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:809]: move this to separate code block and have intro sentence that this create new table with data

File: docs/src/main/sphinx/connector/iceberg.md [hunk 14]
```diff
@@ -1348,33 +1349,40 @@ ORDER BY committed_at DESC
 
 (iceberg-create-or-replace)=
 
-#### Replacing tables
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1351]: Same updates as in the other segment

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 15]
```diff
@@ -780,7 +781,34 @@ directly or used in conditional statements.
 - `$file_size`
   : Size of the file for this row.
 
+### Replace tables
+
+The connector supports replacing an existing table, as
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:786]: I think no comma 

```suggestion
The connector supports replacing an existing table as
```

Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:786]: Also fix wrapping of this paragraph

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 16]
```diff
@@ -780,7 +781,34 @@ directly or used in conditional statements.
...
+an atomic operation. Atomic table replacement creates a new snapshot with the
+new table definition as part of the [table history](#delta-lake-history-table).
+
+To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or
+[`CREATE OR REPLACE AS`](/sql/create-table-as).
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:791]:
```suggestion
[`CREATE OR REPLACE TABLE AS`](/sql/create-table-as).
```

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 17]
```diff
@@ -780,7 +781,34 @@ directly or used in conditional statements.
...
+
+In this example, a table `example_table` can be replaced by a completely new
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:793]:
```suggestion
In this example, a table `example_table` is replaced by a completely new
```

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 18]
```diff
@@ -780,7 +781,34 @@ directly or used in conditional statements.
...
+definition:
+
+```sql
+CREATE TABLE example_table (
+    a BIGINT,
+    b DATE,
+    c BIGINT)
+WITH (partitioning = "partitioned_by");
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:801]: that has to be a column of the table or a function to process the table. See other example

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 18 above)
Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md:801]: @ebyhr?

Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:801]: Either same as other example or just remove it. Having that partitioning defined does nothing to the value of the example

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 19]
```diff
@@ -780,7 +781,34 @@ directly or used in conditional statements.
 - `$file_size`
   : Size of the file for this row.
 
+### Replace tables
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.md:784]: How about moving to before `ALTER TABLE EXECUTE`?

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 20]
```diff
@@ -780,7 +781,34 @@ directly or used in conditional statements.
...
+```
+
+In this example, a new table is created with data:
+
+```sql
+CREATE OR REPLACE TABLE example_table
+WITH (partitioned_by = ARRAY['a'])
+AS SELECT * from another_table;
 (delta-lake-fte-support)=
+```
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.md:811]: This line should be before `(delta-lake-fte-support)=`

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 19 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:784]: Sure .. that works too.

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 20 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:811]: yes.. might be github diff issue or actually wrong

Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md]: This is just weird.

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 21]
```diff
@@ -609,7 +609,28 @@ TABLE AS </sql/create-table-as>` syntax.
 
 The connector supports the following [](/sql/alter-table) statements.
 
+(delta-lake-create-or-replace)=
+
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:613]: remove empty line

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 22]
```diff
@@ -609,7 +609,28 @@ TABLE AS </sql/create-table-as>` syntax.
...
+definition:
+
+```sql
+CREATE OR REPLACE TABLE example_table
+WITH (partitioned_by = ARRAY['a'])
+AS SELECT * from another_table;
+```
+
 (delta-lake-alter-table-execute)=
+
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:633]: remove empty line

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 23]
```diff
@@ -609,7 +609,28 @@ TABLE AS </sql/create-table-as>` syntax.
...
+The connector supports replacing an existing table as an atomic operation.
+Atomic table replacement creates a new snapshot with the new table definition as
+part of the [table history](#delta-lake-history-table).
+
+To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or
+[`CREATE OR REPLACE TABLE AS`](/sql/create-table-as).
+
+In this example, a table `example_table` is replaced by a completely new
+definition:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:624]:
```suggestion
definition and data from the source table.
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 21 above)
Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md:613]: All of the other anchors on the page have a space between the them and the section header.

Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:613]: I know .. we want to get rid of them all eventually

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 24]
```diff
@@ -609,6 +609,25 @@ TABLE AS </sql/create-table-as>` syntax.
...
+To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or
+[`CREATE OR REPLACE TABLE AS`](/sql/create-table-as).
+
+In this example, a table `example_table` is replaced by a completely new
+definition and data from the source table:
+
+```sql
+CREATE OR REPLACE TABLE example_table
+WITH (partitioned_by = ARRAY['a'])
+AS SELECT * from another_table;
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.md:628]: Could you uppercase `from`?. Same for Iceberg example.

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 24 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md]: I fixed that and a few other nits now

**Review by mosabua — APPROVED** [review]
Looks good now. Shipping.

**Pull request merged**